lib/OrdinaryDiffEqCore/src/misc_utils.jl

```diff
@@ -130,3 +130,36 @@ function get_differential_vars(f, u)
 end
 
 isnewton(::Any) = false
+
+
+function _bool_to_ADType(::Val{true}, chunksize, diff_type)
+    Base.depwarn("Using a `Bool` for keyword argument `autodiff` is deprecated. Please use an `ADType` specifier.", :_bool_to_ADType)
+    chunksize = SciMLBase._unwrap_val(chunksize) == 0 ? nothing : SciMLBase._unwrap_val(chunksize)
+    typeof(AutoForwardDiff(chunksize = chunksize))
+end
+
+function _bool_to_ADType(::Val{false}, chunksize, diff_type)
+    Base.depwarn("Using a `Bool` for keyword argument `autodiff` is deprecated. Please use an `ADType` specifier.", :_bool_to_ADType)
+    if diff_type isa Type
+        return typeof(AutoFiniteDiff(fdtype = diff_type()))
+    else
+        return typeof(AutoFiniteDiff(fdtype = diff_type))
+    end
+end
+
+# Functions to get ADType type from Bool or ADType object, or ADType type
+_process_AD_choice(ad_alg::Bool, chunksize, diff_type) = _bool_to_ADType(Val(ad_alg), chunksize, diff_type)
+
+function _process_AD_choice(ad_alg::AbstractADType, chunksize, diff_type) 
+    # need a path for if just chunksize is specified in the Algorithm construction
+    if !(chunksize === Val{0}())
+        return _bool_to_ADType(Val{true}(), chunksize, diff_type)
+    end
```

Comment:
This is dropping the other information?

Reply:
This is in case some one uses the `chunksize` keyword arg, but doesn't use a Bool `autodiff` argument. If they just use `chunksize` the AD used is AutoForwardDiff with that chunksize. 
I should probably have a more specific warning about this route though, especially if they try to use an ADType for `autodiff` and the `chunksize` kwarg at the same time.